harden ChaosMonkey

tbg opened the pull request:
while nodes are restarting, clients (now) return errors
more eagerly. Ignore those (for they're expected). The
log output has become a little more spammy, but I think
that's acceptable for the time being.

Fixes #3450.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3453)

<!-- Reviewable:end -->

tbg committed:
harden ChaosMonkey

while nodes are restarting, clients (now) return errors
more eagerly. Ignore those (for they're expected). The
log output has become a little more spammy, but I think
that's acceptable for the time being.

petermattis commented on acceptance/chaos_test.go:
```diff
@@ -85,7 +84,7 @@ func TestChaos(t *testing.T) {
 					// leader is down. With more realistic retry options, we
 					// should probably not see them.
-					if _, ok := err.(*roachpb.SendError); ok || testutils.IsError(err, rpc.ErrShutdown.Error()) {
+					if _, ok := err.(*roachpb.SendError); ok || testutils.IsError(err, rpc.ErrShutdown.Error()) || testutils.IsError(err, "client is unhealthy") {
```
Perhaps `rpcSender.Send` should return a `SendError` if the client is unhealthy.

tschottdorf replied:
maybe, but that should be in a different change. There are some things to be considered (will this `SendError` be retryable or not? If so, there are infinite retries at `DistSender`). For the purpose of the chaos test, it's convenient to get errors.

tbg committed:
fix TestBuildInfo

it did not wait for gossip to be connected,
which is necessary before the HTTP endpoint
returns anything useful.

petermattis commented on acceptance/chaos_test.go:
```diff
@@ -73,11 +73,10 @@ func TestChaos(t *testing.T) {
 		go func(i int) {
 			r, _ := randutil.NewPseudoRand()
 			value := randutil.RandBytes(r, 8192)
+			k := atomic.LoadInt64(&count)
```
Won't this allow multiple goroutines to write to the same key?

tschottdorf replied:
yes, but that doesn't matter here. We use counts only to track client activity.

petermattis replied:
I'm confused. You're using `k` to generate the key.

tschottdorf replied:
as discussed in person, the values written here don't matter. This is mostly an artifact of this having spawned from `TestPut`.

petermattis commented:
LGTM

BramGruneir commented:
I was working on a different version of this, just having circleci issues last night.
https://github.com/BramGruneir/cockroach/tree/monkey

BramGruneir commented:
LGTM
